src/deriver/queue_manager.py

```
@@ -223,8 +221,25 @@ async def initialize(self) -> None:
         finally:
             await self.cleanup()
 
+    def _on_termination_signal(self, sig: signal.Signals) -> None:
+        """Flip Sentry suppression, then schedule shutdown.
+
+        Suppression is set here rather than in shutdown() because shutdown() is
+        a task: anything raised before it is scheduled would still be reported.
+        """
+        mark_shutting_down()
+        _ = asyncio.create_task(self.shutdown(sig))
```

Ruff (0.16.2)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)